ui/src/MemeGallery.tsx

```diff
@@ -2,19 +2,55 @@ import React, { useState, useEffect } from "react";
 import axios from "axios";
 import "./MemeGallery.css";
 import NavBar from "./NavBar";
+import { Pagination } from "react-bootstrap";
 
 function MemeGallery(props: any) {
   const [memeResponses, setMemeResponses] = useState([]);
+  const [page, setPage] = useState(0);
+  const [memeCount, setMemeCount] = useState(0);
+
+  function getPageCount() {
+    let numberOfPages = Math.ceil(memeCount/10);
+    return numberOfPages;
+  }
+
+  function onPageClick(e: any){
+    console.log('on click', e.target.text);
+    console.log("before " + page);
+    setPage(parseInt(e.target.text));
+    console.log("after " + page);
+
+  }
+
+  function getMemeCount() {
+    axios.get("http://localhost:8080/meme/count").then((response) => {
+      const memeCount = response.data;
+      setMemeCount(memeCount);
+    });
+  }
 
   useEffect(() => {
     function getMemeURLs() {
-      axios.get("http://localhost:8080/meme").then((response) => {
+      axios.get("http://localhost:8080/meme/?page=" + page).then((response) => {
         const memeResponses = response.data;
         setMemeResponses(memeResponses);
       });
     }
     getMemeURLs();
-  }, [props.match.params.id]);
+    getMemeCount();
+  }, [props.match.params.id, page]);
+
+  let pageCount = getPageCount();
+  let items = [];
+  for (let number = 1; number <= pageCount; number++) {
+    items.push(
+      <Pagination.Item key={number} active={number === (page+1)}>
+        {number}
+      </Pagination.Item>,
+    );
+  }
```

Comment:
There's a cleaner way to do this using [Array.from](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/from), passing it a `map` function.